aptos-move/move-examples/fungible_asset/sources/managed_fungible_asset.move

```diff
@@ -0,0 +1,216 @@
+    use aptos_framework::object;
+    use std::error;
+    use std::signer::address_of;
+    use fungible_asset::fungible_source;
+    #[test_only]
+    use fungible_asset::fungible_asset::{create_test_token, balance, is_frozen};
+    #[test_only]
+    use std::signer;
+    #[test_only]
+    use aptos_framework::object::generate_extend_ref;
+
+    /// Mint capability exists or does not exist.
+    const EMINT_CAP: u64 = 1;
+    /// Freeze capability exists does not exist.
+    const EFREEZE_CAP: u64 = 2;
+    /// Burn capability exists or does not exist.
+    const EBURN_CAP: u64 = 3;
+    /// Not the owner.
+    const ENOT_OWNER: u64 = 4;
+    /// Caps existence errors.
+    const EMANAGED_FUNGIBLE_ASSET_CAPS: u64 = 5;
+
+    struct Caps has key {
+        mint: Option<MintCap>,
+        freeze: Option<FreezeCap>,
+        burn: Option<BurnCap>,
+    }
+
+    public fun initialize_managing_capabilities(
+        extend_ref: &ExtendRef,
+        maximum_supply: u64,
+    ) {
+        let (mint_cap, freeze_cap, burn_cap) = init_fungible_source(extend_ref, maximum_supply);
+        let asset_object_signer = object::generate_signer_for_extending(extend_ref);
+        move_to(
+            &asset_object_signer,
+            Caps { mint: option::some(mint_cap), freeze: option::some(freeze_cap), burn: option::some(burn_cap) }
+        )
+    }
+
+    /// Mint fungible tokens as the owner of the base asset.
+    public fun mint_by_asset_owner<T: key>(
+        asset_owner: &signer,
+        asset: &Object<T>,
+        amount: u64,
+        to: address
+    ) acquires Caps {
+        assert_owner(asset_owner, asset);
+        let mint_cap = borrow_mint_from_caps(asset);
+        mint_with_cap(mint_cap, asset, amount, to);
+    }
+
+    /// Burn fungible tokens as the owner of the base asset.
+    public fun burn_by_asset_owner<T: key>(
+        asset_owner: &signer,
+        asset: &Object<T>,
+        amount: u64,
+        from: address
+    ) acquires Caps {
+        assert_owner(asset_owner, asset);
+        let burn_cap = borrow_burn_from_caps(asset);
+        burn_with_cap(burn_cap, asset, amount, from);
+    }
+
+    /// Freeze an owner of fungible asset.
+    public fun freeze_by_asset_owner<T: key>(
+        asset_owner: &signer,
+        asset: &Object<T>,
+        account: address,
+    ) acquires Caps {
+        assert_owner(asset_owner, asset);
+        let freeze_cap = borrow_freeze_from_caps(asset);
+        freeze_with_cap(freeze_cap, account, asset);
+    }
+
+    /// Unfreeze an owner of fungible asset.
+    public fun unfreeze_by_asset_owner<T: key>(
+        asset_owner: &signer,
+        asset: &Object<T>,
+        fungible_asset_owner: address
+    ) acquires Caps {
+        assert_owner(asset_owner, asset);
+        let freeze_cap = borrow_freeze_from_caps(asset);
+        unfreeze_with_cap(freeze_cap, fungible_asset_owner, asset);
+    }
+
+    public fun owner_can_mint<T: key>(asset: &Object<T>): bool acquires Caps {
```

Comment:
May be these can become view functions if view functions can accept object references. cc @gerben-stavenga

Reply:
Yes, we are missing `Object<T>` as an input to entry.